Python/sysmodule.c

```diff
@@ -3319,6 +3320,14 @@ make_impl_info(PyObject *version_info)
     if (res < 0)
         goto error;
 
+    value = PyLong_FromLong(PYC_MAGIC_NUMBER);
+    if (value == NULL)
+        goto error;
+    res = PyDict_SetItemString(impl_info, "pyc_magic_number", value);
```

Comment:
I'm wondering if `sys.implementation` is the best place for this? https://docs.python.org/3/library/sys.html#sys.implementation has all attributes being required for all implementations and bytecode is not necessary to implement Python. Should it just be a private attribute in sys instead?

Reply:
According to those docs, this can't be added as `sys.implementation.pyc_magic_number` [without a PEP](https://github.com/python/cpython/blob/1d607fe759ef22177b50d734ae029df3903c99e0/Doc/library/sys.rst?plain=1#L1136-L1137).  `sys.implementation._pyc_magic_number` (left undocumented) [would be fine](https://github.com/python/cpython/blob/1d607fe759ef22177b50d734ae029df3903c99e0/Doc/library/sys.rst?plain=1#L1125-L1127), though.

Reply:
I've made this private.